2.6 - Error with config locations

### Describe the bug

I have an `application.yaml` that gets used for all projects (containing things like `ignored-split-packages`). With the `ConfigGenerationBuildStep` changes in 2.6 I now get an exception during builds because of the following lines in my `build.gradle` file.

```groovy
    build {
        // Shared Quarkus configuration applied to all projects
        def shareConfig = rootProject.projectDir.toURI().toString() + "/conf/quarkus"
        System.setProperty("quarkus.config.locations", "${shareConfig}")
    }
```

It looks like `ConfigGenerationBuildStep` doesn't seem to like the URI string format (see exception below).

 I also tried  `toAbsolutePath()` instead of `toURI().toString()`, but then it pushed the error to `bstractLocationConfigSourceLoader`, which didn't like the drive letter when in Windows (see exception below).

### Expected behavior

Successful build

### Actual behavior

Error using `toURI().toString()`
```
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 4: file:/C:/my/git//conf/quarkus
        at io.quarkus.deployment.steps.ConfigGenerationBuildStep.lambda$watchConfigFiles$0(ConfigGenerationBuildStep.java:259)
        at io.quarkus.deployment.steps.ConfigGenerationBuildStep.watchConfigFiles(ConfigGenerationBuildStep.java:257)
```

Error using `toAbsolutePath()`
```
Caused by: java.lang.IllegalArgumentException: SRCFG00033: Scheme C not supported
        at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:101)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:80)
        at io.smallrye.config.AbstractLocationConfigSourceFactory.getConfigSources(AbstractLocationConfigSourceFactory.java:30)
        at io.smallrye.config.PropertiesLocationConfigSourceFactory.getConfigSources(PropertiesLocationConfigSourceFactory.java:16)
        at io.smallrye.config.ConfigurableConfigSource.unwrap(ConfigurableConfigSource.java:53)
        at io.smallrye.config.ConfigurableConfigSource.getConfigSources(ConfigurableConfigSource.java:49)
        at io.smallrye.config.SmallRyeConfig$ConfigSources.mapLateSources(SmallRyeConfig.java:618)
        at io.smallrye.config.SmallRyeConfig$ConfigSources.<init>(SmallRyeConfig.java:529)
        at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:66)
        at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:419)
        at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:173)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:108)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:328)
```

### How to Reproduce?

```groovy
    build {
        // Shared Quarkus configuration applied to all projects
        def shareConfig = rootProject.projectDir.toURI().toString() + "/conf/quarkus"
        System.setProperty("quarkus.config.locations", "${shareConfig}")
    }
```

or

```groovy
    build {
        // Shared Quarkus configuration applied to all projects
        def shareConfig = rootProject.projectDir.getAbsolutePath() + "/conf/quarkus"
        System.setProperty("quarkus.config.locations", "${shareConfig}")
    }
```

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-19042

### Output of `java -version`

OpenJDK 11.0.11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.3.3

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java
@@ -10,7 +10,9 @@ import static java.util.stream.Collectors.toSet;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.lang.reflect.Modifier;
+import java.net.URI;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Collections;
@@ -253,12 +255,13 @@ public class ConfigGenerationBuildStep {
         configWatchedFiles.add(
                 Paths.get(userDir, "config", String.format("application-%s.properties", profile)).toAbsolutePath().toString());
 
-        Optional<List<String>> optionalLocations = config.getOptionalValues(SMALLRYE_CONFIG_LOCATIONS, String.class);
+        Optional<List<URI>> optionalLocations = config.getOptionalValues(SMALLRYE_CONFIG_LOCATIONS, URI.class);
         optionalLocations.ifPresent(locations -> {
-            for (String location : locations) {
-                if (!Files.isDirectory(Paths.get(location))) {
-                    configWatchedFiles.add(location);
-                    configWatchedFiles.add(appendProfileToFilename(location, profile));
+            for (URI location : locations) {
+                Path path = location.getScheme() != null ? Paths.get(location) : Paths.get(location.getPath());
+                if (!Files.isDirectory(path)) {
+                    configWatchedFiles.add(location.toString());
+                    configWatchedFiles.add(appendProfileToFilename(location.toString(), profile));
                 }
             }
         });

--- a/core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java
@@ -1,5 +1,6 @@
 package io.quarkus.runtime;
 
+import java.net.URI;
 import java.util.List;
 import java.util.Optional;
 
@@ -21,7 +22,7 @@ public class ConfigConfig {
      * separated by a comma and each must represent a valid {@link java.net.URI}.
      */
     @ConfigItem(name = "config.locations")
-    public Optional<List<String>> locations;
+    public Optional<List<URI>> locations;
 
     /**
      * Accepts a single configuration profile name. If a configuration property cannot be found in the current active

